src/user.ts

```diff
@@ -120,6 +120,25 @@ export class User {
         await this.storage.username;
     }
 
+    /**
+     * @param {string} username - a string param that represents the username
+     * @returns {boolean} true if username already exists and false if doesn't exists
+     */
+
+    async usernameExists(username: string): Promise<boolean> {
+        const normalizedUsername = username.normalize('NFKC');
+
+        const fullUsername = TonomyUsername.fromUsername(
+            normalizedUsername,
+            AccountType.PERSON,
+            getSettings().accountSuffix
+        );
+
+        (await User.getAccountInfo(fullUsername)) as any; // Throws error if username is taken
```

Comment:
what if this throws a different error? better to put this in a try catch block, and check the error type. If the error is that the username is taken, then return false, otherwise throw the error